core/string_api.php

```diff
@@ -445,13 +448,15 @@ function( $p_array ) use( $p_detail_info, $p_fqdn ) {
 		} else {
 			$s_bugnote_link_callback[$p_include_anchor][$p_detail_info][$p_fqdn] =
 				function( $p_array ) {
-					$t_bug_id = bugnote_get_field( (int)$p_array[2], 'bug_id' );
-					if( $t_bug_id && bug_exists( $t_bug_id ) ) {
-						return $p_array[1] .
-							string_get_bugnote_view_url_with_fqdn( $t_bug_id, (int)$p_array[2] );
-					} else {
-						return $p_array[0];
+					$c_bugnote_id = (int)$p_array[2];
+					if( bugnote_exists( $c_bugnote_id ) ) {
+						$t_bug_id = bugnote_get_field( $c_bugnote_id, 'bug_id' );
+						if( $t_bug_id && bug_exists( $t_bug_id ) ) {
+							return $p_array[1] .
+								string_get_bugnote_view_url_with_fqdn( $t_bug_id, $c_bugnote_id );
+						}
 					}
+					return $p_array[0];
```

Comment:
Moving return $p_array[0] outside of else branches is valid, since this should be the default (unmodified) string. Other positive conditions finalize with its own "return"

But i added a check for bugnote_exists that wasn't there in the first place. I will try to verify that i can find a test case for that code path (not sure right now if i tested it explicitly)